tests/test_xlora.py

```diff
@@ -0,0 +1,172 @@
+# Copyright 2023-present the HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+import torch
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+from peft import LoraConfig, PeftType, TaskType, XLoraConfig, get_peft_model
+
+
+model_id = "facebook/opt-125m"
+tokenizer = AutoTokenizer.from_pretrained(model_id, trust_remote_code=True, device_map="cuda:0")
+tmp_path = "/tmp/xlora_test"
+os.makedirs(tmp_path, exist_ok=True)
+
+for i in range(1, 9):
+    torch.manual_seed(i)
+    lora_config = LoraConfig(task_type="CAUSAL_LM", init_lora_weights=False)
+    model = AutoModelForCausalLM.from_pretrained(model_id)
+    peft_model = get_peft_model(model, lora_config)
+    peft_model.save_pretrained(f"{tmp_path}/checkpoint-{i}")
+    print(f"finished {i} of 8")
+
+
+class TestXlora:
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+    def test_functional(self):
+        tokenizer = AutoTokenizer.from_pretrained(model_id, trust_remote_code=True, device_map="cuda:0")
+
+        model = AutoModelForCausalLM.from_pretrained(model_id)
```

Comment:
Similar to my snippet above, we could turn the model into a fixture and use that for each test function that needs it. However, we cannot make it class-scoped because the test functions mutate it.

Reply:
I think this is a great idea, and I think it is absolutely doable! I will implement it shortly.